### DISCUSSION TITLE
项目里配置是如何动态映射nacos里的配置的？

### DISCUSSION THREAD

**Discussion by zqb666** [zuihou/zuihou-admin-cloud#247] [02.提问] [answered]
<img width="620" alt="image" src="https://user-images.githubusercontent.com/6092525/226273489-30cdf277-9335-4d6d-9d45-587f3fbc1d14.png">
比如这些配置 通过一定的表达式 是怎么映射到nacos 的具体配置的 这一块的代码在哪？

**Comment by zuihou** [comment]
https://github.com/dromara/lamp-cloud/discussions?discussions_q=nacos.ip

**Comment by zuihou** [comment]
https://github.com/dromara/lamp-cloud/discussions/58

**Reply by zqb666** [reply]
<img width="536" alt="image" src="https://user-images.githubusercontent.com/6092525/226275509-12f6bc0c-ecec-4fce-9316-f11c6bba4162.png">
问题1，这块是怎么对应起来的 实现原理是什么？
问题2，我在后面加 不是取配置的配置能生效吗？
<img width="1030" alt="image" src="https://user-images.githubusercontent.com/6092525/226275937-ef792ded-2b44-4e23-8f79-11a8c34d11d1.png">

问题3，最后再问下 我 redis加了集群配置
<img width="991" alt="image" src="https://user-images.githubusercontent.com/6092525/226276199-8141f810-ce13-41ad-b906-9114f4566b06.png">
lamp-cloud 里 lamp-util里的redis的配置类需要修改成集群的配置吗 还是目前这个就能自动适配集群
麻烦群主 解答下我的疑惑 十分感谢
🙏

**Comment by zuihou** [comment, marked as answer]
问题1:
17-20行中的 ${xxx} 读取的是项目配置文件中定义的 xxx 属性。
也就是说xxx定义在本文件中 或者 其他配置文件中，只要能被项目正常加载，都能识别。

这个是spring boot 提供的功能。

**Comment by zuihou** [comment]
问题2： 
可以写自己的配置，无需采用 ${} 占位符， 但要注意格式。
yml文件是有强烈的格式要求的，缩进是2个空格。  

你这个缩进明显不对。

**Comment by zuihou** [comment]
问题3： lamp-util 中 lamp-cache-starter 并没有对redis的配置做限制， 无论你是 单机还是集群，只要按照 spring-boot-starter-data-redis 的要求和格式来配置就行。


不会配置，请搜索 spring-boot-starter-data-redis 如何配置集群
